File: Core/Entities/Product.cs
using System;
using System.Collections.Generic;

namespace Core.Entities
{
    public class Product: BaseEntity
    {
        public string Name { get; set ;}
        public decimal Price { get; set; }
        public string Description { get; set; }
        public DateTime DateAdded { get; set; }


        //Rodzaj produktu
        public ProductType ProductType { get; set; }
        public int ProductTypeId { get; set; }
        public ProductRoom ProductRoom { get; set; } 
        public int ProductRoomId { get; set; }

        //Szczegóły produktu
        public int Height { get; set; }
        public int Width { get; set; }
        public string Color1 { get ; set; }
        public string Color2 { get; set; }
        public string Material1 { get; set; }
        public string Material2 { get; set; }


        //Zdjęcia produktu
        public ICollection<ProductPhotos> Photos { get; set; }


    }
}

File: SPA/src/app/shared/models/product.ts
export interface IProduct{
  id: number;
  name: string;
  price: number;
  description: string;
  dateAdded: string;
  productType: string;
  productRoom: string;
  productTypeId: number;
  productRoomId: number;
  height: number;
  width: number;
  color1: string;
  color2: string;
  material1: string;
  material2: string;
  pictureUrl: string;
  photos: IProductPhotos[];
}

export interface IProductPhotos {
  id: number;
  pictureUrl: string;
}

export interface IProductToEdit {
  
  name: string;
  price: number;
  description: string;
  productTypeId: number;
  productRoomId: number;
  height: number;
  width: number;
  color1: string;
  color2: string;
  material1: string;
  material2: string;
}

